tests/utils/iotrace.py

```diff
@@ -68,22 +75,22 @@ def stop_tracing(self) -> bool:
         :raises Exception: if could not stop tracing which is active
         '''
         TestRun.LOGGER.info("Stopping tracing")
-        pid = TestRun.executor.run('pgrep iotrace')
-        if pid.stdout == "":
-            return False
 
         # Send sigints
         kill_attempts = 30
         attempt = 0
-        while pid.stdout != "" and attempt < kill_attempts:
-            TestRun.LOGGER.info("Sending sigint no. " + str(attempt))
+        iotrace_active = self.check_if_tracing_active()
+        while iotrace_active and attempt < kill_attempts:
+            TestRun.LOGGER.info("Sending sigint no. " + str(attempt + 1))
             attempt += 1
-            TestRun.executor.run(f'kill -s SIGINT {pid.stdout}')
+            TestRun.executor.run(f'kill -s SIGINT {self.pid}')
             time.sleep(2)
-            pid = TestRun.executor.run('pgrep iotrace')
+            iotrace_active = self.check_if_tracing_active()
 
-        if pid.stdout != "":
-            raise Exception("Could not kill iotrace")
+        self.pid = TestRun.executor.run('pgrep iotrace').stdout
+        if self.pid != "":
+            TestRun.LOGGER.error("Could not kill iotrace")
+            return False
```

Comment:
There is assumption, that two instances of tracer can't run simultaneously, so I wouldn't add this check in path of regular execution.